networking/kvm.go

```diff
@@ -113,32 +220,87 @@ func kvmSetup(podRoot string, podID types.UUID, fps []ForwardedPort, privateNetL
 			}
 
 			// add address to host tap device
-			err = netlink.AddrAdd(
+			err = ensureHasAddr(
 				link,
-				&netlink.Addr{
-					IPNet: &net.IPNet{
-						IP:   n.runtime.HostIP,
-						Mask: net.IPMask(n.runtime.Mask),
-					},
-					Label: ifName,
-				})
+				&net.IPNet{
+					IP:   n.runtime.IP4.Gateway,
+					Mask: net.IPMask(n.runtime.Mask),
+				},
+			)
 			if err != nil {
 				return nil, fmt.Errorf("cannot add address to host tap device %q: %v", ifName, err)
 			}
 
-			if n.conf.IPMasq {
-				h := sha512.Sum512([]byte(podID.String()))
-				chain := fmt.Sprintf("CNI-%s-%x", n.conf.Name, h[:8])
-				if err = ip.SetupIPMasq(&net.IPNet{
-					IP:   n.runtime.IP,
-					Mask: net.IPMask(n.runtime.Mask),
-				}, chain); err != nil {
-					return nil, err
+			time.Sleep(time.Second / 2) // HACKHACKHACK - without this delay below changes in routes are not persistent
```

Comment:
If You have other solution to this problem - please, provide it.

I think that adding issue before merge is premature, because this code could be rejected by repo owners and in such case issue would be incorrect.